corehq/ex-submodules/casexml/apps/phone/xml.py

```diff
@@ -129,10 +130,16 @@ def get_registration_element_data(restore_user):
         "password": restore_user.password,
         "uuid": restore_user.user_id,
         "date": date_to_xml_string(restore_user.date_joined),
-        "user_data": restore_user.user_session_data
+        "user_data": get_session_data_with_abdm_token(restore_user)
```

Comment:
Is it possible to check for a particular application, rather than a domain? For example, if some property is set within an application, then only the key will be returned. 

Will check the extensions part.

Reply:
> Is it possible to check for a particular application, rather than a domain? For example, if some property is set within an application, then only the key will be returned.

That could be possible. You would need to change the `get_registration_element` to accept more arguments (potentially just pass in the whole `RestoreState` class)